targets/csharp/source/Plugins/CloudScript/source/PlayFabFunctionContexts.cs

```diff
@@ -0,0 +1,245 @@
+    /// Meant to be used with task based Azure Functions.
+    /// Provides  fields through ApiSettings and AuthenticationContext, the profile of the caller, as well as the 
+    /// arguments passed to the execution request.
+    /// </summary>
+    /// <typeparam name="TFunctionArgument">Type of FunctionArgument</typeparam>
+    public class FunctionTaskContext<TFunctionArgument>
+    {
+        public PlayFabApiSettings ApiSettings { get; private set; }
+
+        public PlayFabAuthenticationContext AuthenticationContext { get; set; }
+
+        public NameIdentifier ScheduledTaskNameId { get; set; }
+
+        public Stack<PlayStreamEventHistory> EventHistory { get; set; }
+
+        public TFunctionArgument FunctionArgument { get; set; }
+
+        protected FunctionTaskContext() { }
+
+        /// <summary>
+        /// Creates a new <c>FunctionTaskContext</c> out of the incoming request to an Azure Function.
+        /// </summary>
+        /// <param name="request">The request incoming to an Azure Function from the PlayFab server</param>
+        /// <returns>A new populated <c>FunctionTaskContext</c> instance</returns>
+        public static async Task<FunctionTaskContext<TFunctionArgument>> Create(HttpRequestMessage request)
+        {
+            using (var content = request.Content)
+            {
+                var body = await content.ReadAsStringAsync();
+                var contextInternal = PlayFabSimpleJson.DeserializeObject<FunctionTaskContextInternal>(body);
+                var settings = new PlayFabApiSettings
+                {
+                    TitleId = contextInternal.TitleAuthenticationContext.Id,
+                    DeveloperSecretKey = Environment.GetEnvironmentVariable("PLAYFAB_DEV_SECRET_KEY", EnvironmentVariableTarget.Process)
+                };
+
+                var authContext = new PlayFabAuthenticationContext
+                {
+                    EntityToken = contextInternal.TitleAuthenticationContext.EntityToken
+                };
+
+                return new FunctionTaskContext<TFunctionArgument>()
+                {
+                    ApiSettings = settings,
+                    AuthenticationContext = authContext,
+                    ScheduledTaskNameId = contextInternal.ScheduledTaskNameId,
+                    EventHistory = contextInternal.EventHistory,
+                    FunctionArgument = contextInternal.FunctionArgument
+                };
+            }
+        }
+
+        private class FunctionTaskContextInternal
+        {
+            public NameIdentifier ScheduledTaskNameId { get; set; }
+
+            public Stack<PlayStreamEventHistory> EventHistory { get; set; }
+
+            public TitleAuthenticationContext TitleAuthenticationContext { get; set; }
+
+            public TFunctionArgument FunctionArgument { get; set; }
+        }
+    }
+
+    public class NameIdentifier
+    {
+        public string Name { get; set; }
+
+        public string Id { get; set; }
```

Comment:
What is this Id? EntityId?

Reply:
I'm not sure. It's part of PlayStream types and I created a 1:1 of that type here as it's needed for my deserialization until we get type generation going later.

Reply:
Since it's unclear, it would be nice if you'd find out and add a summary comment for it.